### PR TITLE
Implemented drawing images on BgrRasterCanvas

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Implemented `draw ~image` for `BgrRasterCanvas`.
@marcusnaslund ready for review

### CODE REVIEW

File: source/draw/RasterBgr.ooc [hunk 1]
```diff
@@ -32,6 +32,22 @@ BgrRasterCanvas: class extends RasterCanvas {
...
 	}
+	draw: override func ~ImageSourceDestination (image: Image, source, destination: IntBox2D) {
+		bgr: RasterBgr = null
+		if (image instanceOf?(This)) {
+			bgr = image as RasterBgr
+			bgr referenceCount increase()
+		} else if (image instanceOf?(RasterImage))
+			bgr = RasterBgr convertFrom(image as RasterImage)
+		if (bgr) {
+			if (this target size == bgr size && this target stride == bgr stride && source == destination && source size == bgr size && source leftTop x == 0 && source leftTop y == 0)
```
Comment (sebastianbaginski) [inline source/draw/RasterBgr.ooc]: I know it looks scary, but we really want `memcpy` when we can avoid per pixel loops.

File: source/draw/RasterBgr.ooc [hunk 2]
```diff
@@ -62,13 +78,8 @@ RasterBgr: class extends RasterPacked {
...
-		result: This
-		if (this size == size)
-			result = this copy()
-		else {
-			result = This new(size)
-			RasterCanvas resizeNearestNeighbour(this buffer pointer as ColorBgr*, result buffer pointer as ColorBgr*, IntBox2D new(this size), IntBox2D new(result size), this stride, result stride, this bytesPerPixel)
-		}
+		result := This new(size)
+		result canvas draw(this, IntBox2D new(this size), IntBox2D new(size))
 		result
```
Comment (sebastianbaginski) [inline source/draw/RasterBgr.ooc]: In the future this function will be moved directly to `Image` class. For now not every Canvas subclass can draw images.

**Comment by marcusnaslund** [comment]
Segfault.

**Comment by sebastianbaginski** [comment]
:/ not this again

**Comment by sebastianbaginski** [comment]
Can you try again now ?

**Comment by marcusnaslund** [comment]
I did not try it, CircleCI segfaulted. Let's see now.

**Comment by marcusnaslund** [comment]
Yes. :)

**Pull request merged**